app/code/Magento/ImportExport/Model/Import/Source/FileParser/CorruptedFileException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Model\Import\Source\FileParser;
+
+class CorruptedFileException extends \RuntimeException
+{
+    private $fileName;
+
+    public function __construct($fileName = '', $message = '')
+    {
+        $this->fileName = $fileName;
+        parent::__construct($message ?: sprintf('File "%s" is corrupted', $fileName));
+    }
+
+
+    public function getFileName()
+    {
+        return $this->fileName;
+    }
```

Comment:
The file name can be of a help for client code that catches this specific exception. I can remove it if filename is redundant error information for parsing files.